src/app/app.component.ts

```diff
@@ -23,3 +27,69 @@ export class AppComponent {
 
   }
 }
+
+@Component({
+  selector: 'app-calendar-header',
+  template: `
+    <mat-select placeholder="Calendar Format" [formControl]="form.controls['calendar']">
+      <mat-option *ngFor="let cal of supportedCalendarFormats" [value]="cal">{{cal}}</mat-option>
+    </mat-select>
+    <mat-calendar-header></mat-calendar-header>
```

Comment:
@mmalerba This works now :-) (https://github.com/angular/material2/pull/10856)

Reply:
@mmalerba done in https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/pull/6/commits/00231b6a823169e3520997c79c2e2376a968b903

> If you want to send a PR adding a updateTodaysDate() method to the datepicker, I would be ok with that. This solution seems kind of messy.

I agree. What about a general method to do a clean reinit? Once we add more calendar formats, the names of the months etc. may be completely different, so we need to reinit the whole thing.